crates/volta-core/src/manifest/mod.rs

```diff
@@ -123,7 +114,26 @@ impl Manifest {
     }
 }
 
-// unit tests
+pub struct BinManifest {
+    /// The `bin` section, containing a map of binary names to locations.
+    pub bin: HashMap<String, String>,
+    /// The `engines` section, containing a spec of the Node versions that the package works on.
+    pub engine: Option<String>,
+}
+
+impl BinManifest {
+    pub fn for_dir(project_root: &Path) -> Fallible<Self> {
+        let package_file = project_root.join("package.json");
+        let file = File::open(&package_file).with_context(|_| ErrorDetails::PackageReadError {
+            file: package_file.to_path_buf(),
+        })?;
 
-#[cfg(test)]
-pub mod tests;
+        let serial: serial::RawBinManifest =
+            serde_json::de::from_reader(file).with_context(|_| {
+                ErrorDetails::PackageParseError {
+                    file: package_file.to_path_buf(),
+                }
+            })?;
+        Ok(serial.into_bin_manifest())
```

Comment:
FWIW, you can make this a little shorter and IMO clearer:

```rust
serde_json::de::from_reader(file)
    .with_context(|_| ErrorDetails::PackageParseError {
        file: package_file.to_path_buf(),
    })
    .map(serial::RawBinManifest::into_bin_manifest)
```

Reply:
Done!